run

```diff
@@ -141,9 +148,11 @@ c_str() {
 }
 
 c_echo() {
+    # shellcheck disable=SC2155
     local old_opt="$(shopt -op xtrace)" # save old xtrace option
```

Comment:
Guessing breaking out `local old_opt` on a prior line causes the `shopt` call not to work. Is that right?